data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/BufferAccumulator.java

```diff
@@ -0,0 +1,130 @@
+/*
```

Comment:
Could we pull out the original BufferAccumulator into a place where multiple plugins can consume it? It'd be better to maintain the logic in a singular place

Reply:
Agreed but can we do that in the future? I’ll create an issue for it

Reply:
Here is the followup PR after this one is merged (https://github.com/opensearch-project/data-prepper/pull/2857)